### PR TITLE
Fixed the CloudFormation template to complete the Jenkins setup

### PR DESCRIPTION
**Pull request by christianhxc**
*Issue #, if available:* #157

*Description of changes:*
The existing CloudFormation template is installing the most recent version of Jenkins. The problem is that the userdata script is restoring a Jenkins back from an older version. I changed the install instructions to specify the Jenkins version. The one that the backup has is `2.138.4`, and because it's a very old version, I had to add the `--nogpgcheck`. We should work on update the workshop to use a more recent version to avoid this problem.

I also had to change the instructions to download the CloudFormation template from GitHub and not from S3.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by jagpk** [comment]
image

Health for Jenkins Master is failing. Please test at your end confirm if it works.

File: content/amazon-ec2-spot-cicd-workshop/prep.md [hunk 1]
```diff
@@ -19,14 +19,14 @@ You will need to access the SSH interfaces of some Linux EC2 instances created i
 ## LAUNCH THE CLOUDFORMATION TEMPLATE
 So that you can concentrate on the aspects of this workshop that directly relate to Amazon EC2 Spot instances, there is a CloudFormation template that will deploy the base AWS infrastructure needed for all of the labs within the workshop - saving you from having to create things like VPCs, Security Groups, IAM policies and so forth.
 
-Deploy the CloudFormation template located at:
-[https://s3-us-west-2.amazonaws.com/amazon-ec2-spot-cicd-workshop/amazon-ec2-spot-cicd-workshop.yaml](https://s3-us-west-2.amazonaws.com/amazon-ec2-spot-cicd-workshop/amazon-ec2-spot-cicd-workshop.yaml)
+Download and deploy the CloudFormation template located at:
+[https://raw.githubusercontent.com/awslabs/ec2-spot-workshops/master/workshops/amazon-ec2-spot-cicd-workshop/amazon-ec2-spot-cicd-workshop.yaml](https://raw.githubusercontent.com/awslabs/ec2-spot-workshops/master/workshops/amazon-ec2-spot-cicd-workshop/amazon-ec2-spot-cicd-workshop.yaml)
```
[the remark this answers is not in the document]
Comment (christianhxc) [reply content/amazon-ec2-spot-cicd-workshop/prep.md]: ok, done

File: content/amazon-ec2-spot-cicd-workshop/prep.md [hunk 2]
```diff
@@ -19,14 +19,14 @@ You will need to access the SSH interfaces of some Linux EC2 instances created i
 ## LAUNCH THE CLOUDFORMATION TEMPLATE
 So that you can concentrate on the aspects of this workshop that directly relate to Amazon EC2 Spot instances, there is a CloudFormation template that will deploy the base AWS infrastructure needed for all of the labs within the workshop - saving you from having to create things like VPCs, Security Groups, IAM policies and so forth.
 
-Deploy the CloudFormation template located at:
-[https://s3-us-west-2.amazonaws.com/amazon-ec2-spot-cicd-workshop/amazon-ec2-spot-cicd-workshop.yaml](https://s3-us-west-2.amazonaws.com/amazon-ec2-spot-cicd-workshop/amazon-ec2-spot-cicd-workshop.yaml)
+Download and deploy the CloudFormation template located at:
```
[the remark this answers is not in the document]
Comment (christianhxc) [reply content/amazon-ec2-spot-cicd-workshop/prep.md]: ok, done

File: workshops/amazon-ec2-spot-cicd-workshop/amazon-ec2-spot-cicd-workshop.yaml [hunk 3]
```diff
@@ -301,14 +300,14 @@ Resources:
...
+              Service:
+                - ecs.amazonaws.com
+            Action:
+              - sts:AssumeRole
       ManagedPolicyArns:
-      - arn:aws:iam::aws:policy/service-role/AmazonEC2ContainerServiceRole
+        - arn:aws:iam::aws:policy/service-role/AmazonEC2ContainerServiceRole
       Path: /
 
   ECSServiceLinkedRole:
```
[the remark this answers is not in the document]
Comment (christianhxc) [reply workshops/amazon-ec2-spot-cicd-workshop/amazon-ec2-spot-cicd-workshop.yaml:313]: I'll come back at this later

File: workshops/amazon-ec2-spot-cicd-workshop/amazon-ec2-spot-cicd-workshop.yaml [hunk 4]
```diff
@@ -604,26 +603,26 @@ Resources:
...
-        FromPort: 2049
-        ToPort: 2049
-        SourceSecurityGroupId: !Ref SecurityGroupJenkins
+        - IpProtocol: tcp
+          FromPort: 2049
+          ToPort: 2049
+          SourceSecurityGroupId: !Ref SecurityGroupJenkins
       VpcId: !Ref VPC
 
   AMILookupLambdaFunction: # A Lambda function that will be used to look up the AMI for the latest Amazon Linux AMI. The source code for this Lambda function can be obtained at https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/walkthrough-custom-resources-lambda-lookup-amiids.html.
```
[the remark this answers is not in the document]
Comment (christianhxc) [reply workshops/amazon-ec2-spot-cicd-workshop/amazon-ec2-spot-cicd-workshop.yaml:628]: I'll come back at this later

**Pull request merged**